docs/api-reference/index.md

```diff
@@ -19,6 +19,10 @@
    :recursive:
 
    providers
+   set_background_runs
```

Comment:
As was suggested in https://github.com/scipp/essdiffraction/pull/63, should we rename to `with_*`?

Otherwise, everything looks good :+1:

Reply:
Done